feat(screenshot): Add customizable color option for masked elements

Darkensses opened the pull request:
I added a new option to the screenshot method to customize the color of the box when we want to mask some elements for the screenshot.

The default color is pink `#FF00FF`, but with this new option you can specify the color you like the most, like a nice green `#00FF00`:

```js
await page.screenshot({
  mask: [page.locator('div').nth(5)],
  maskColor: "#00FF00",
})
```

![ss](https://github.com/microsoft/playwright/assets/23271049/05f754de-0ba6-47a3-ae3e-769720d3da3b)

Darkensses committed:
docs: Add maskColor option for screenshot method

Darkensses committed:
test(screenshot): mask color is different than pink

Darkensses committed:
feat(screenshot): Add customizable color option for masked elements

Darkensses commented:
> @microsoft-github-policy-service agree

@microsoft-github-policy-service agree

Darkensses committed:
chore(lint): Update test for the new maskColor option

Changed the doublequote to singlequote

Signed-off-by: Jasiel Guillén <darkensses@gmail.com>

Darkensses committed:
chore: Restore space in update params.md

There was a space that was remove after saving the file locally. This commit restore it.

Signed-off-by: Jasiel Guillén <darkensses@gmail.com>

dgozman commented on docs/src/api/params.md:
```diff
@@ -1135,6 +1140,7 @@ When set to `"hide"`, screenshot will hide text caret. When set to `"initial"`,
 - %%-screenshot-option-caret-%%
 - %%-screenshot-option-type-%%
 - %%-screenshot-option-mask-%%
+- %%-screenshot-option-mask-color-%%
```
Since this option was not added in v1.8, you'll have to manually add it everywhere `screenshot-options-common-list-v1.8` is mentioned, remove it from this list, and insert `since: v1.34` before line 1096.

dgozman reviewed (commented):
Thank you for the PR! Looks pretty good, just a few comments.

Darkensses committed:
chore(screenshot): Added manually the new maskColor option

https://github.com/microsoft/playwright/pull/23185#pullrequestreview-1437164520

Darkensses committed:
test(screenshot): Added three expected pngs for each browser

https://github.com/microsoft/playwright/pull/23185#discussion_r1200852306

Darkensses commented:
> Thank you for the PR! Looks pretty good, just a few comments.

Done! I made changes to the files according to your suggestions. Let me know if you need something else :) thanks!

dgozman reviewed (approved):
Looks great, merging in.
